src/cli-client/snap/snapcraft.yaml

```diff
@@ -0,0 +1,31 @@
+name: oci-factory # TODO 'snapcraft register <name>'
+base: core22
+version: '0.0.1'
+summary: The OCI Factory CLI client to build, upload and release OCI images
+description: |
+  The OCI Factory CLI client is a tool that builds, tests, and releases
+  the OCI images owned by Canonical using the Github workflow in 
+  the OCI Factory repository.
+grade: devel # must be 'stable' to release into candidate/stable channels
+confinement: strict
+
+parts:
+  oci-factory:
+    plugin: go
+    build-snaps: [go/latest/stable]
+    source: https://github.com/canonical/oci-factory
+    source-type: git
+    source-subdir: src/cli-client/cmd/oci-factory
+    stage-packages:
+      - git
+    # TODO remove this override-pull when the branch is merged
+    override-pull: |
+      snapcraftctl pull
+      git checkout ROCKS-1216-cli-client
```

Comment:
I guess for this :p which I also guess is a leftover?

Reply:
I don't think we already have the CLI client in the default branch of `oci-factory` repo. So I have to add this `override-pull` to test the snapcraft packaging before this branch is merged into main.

Reply:
sure but that's for testing. it shouldn't go to `main` with this branch name

Reply:
well, you should actually be using `.` as your source. that way you don't need this.

Reply:
Yeah, good suggestion.